sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1060,7 +1087,7 @@ protected void cleanupTempResource(BigQueryOptions bqOptions) throws Exception {
     @Override
     public void populateDisplayData(DisplayData.Builder builder) {
       super.populateDisplayData(builder);
-      builder.add(DisplayData.item("query", query.get()));
+      builder.add(DisplayData.item("query", tableProvider(query)));
```

Comment:
This seems a bit odd because a query isn't a table. Maybe this should just be something like `stringProviderToString(...)`? I could also see putting this method on ValueProvider. Something like `valueOrProviderToString` which returns `isAccessible() ? get().toString() : toString()`